CONTRIBUTING.md

```diff
@@ -29,26 +31,119 @@ The [PyScript Code of Conduct](https://github.com/pyscript/governance/blob/main/
 
 # Contributing
 
-## Reporting bugs
+## Opening an Issue
+
+If you encounter a bug, have a feature proposal or have a question, please open an issue on the PyScript repository and select the right category. Feature proposals and questions will be opened as a `Discussion` to encourage discussions around the topic, while bugs are filed under traditional Github Issues. When opening an issue, make sure to include the following information:
+
+-   A clear and concise title that summarizes the issue.
+-   A detailed description of the problem or feature request.
+-   Steps to reproduce the issue (if applicable).
+-   Any relevant error messages or screenshots.
+-   The version of PyScript you are using (if applicable).
+
+### Reporting bugs
 
 Bugs are tracked on the [project issues page](https://github.com/pyscript/pyscript/issues). Please check if your issue has already been filed by someone else by searching the existing issues before filing a new one. Once your issue is filed, it will be triaged by another contributor or maintainer. If there are questions raised about your issue, please respond promptly.
 
-## Creating useful issues
+### Creating useful issues
 
 -   Use a clear and descriptive title.
 -   Describe the specific steps that reproduce the problem with as many details as possible so that someone can verify the issue.
 -   Describe the behavior you observed, and the behavior you had expected.
 -   Include screenshots if they help make the issue clear.
 
-## Reporting security issues
+### Reporting security issues
 
 If you aren't confident that it is appropriate to submit a security issue using the above process, you can e-mail it to security@pyscript.net
 
-## Asking questions
+### Asking questions
 
 If you have questions about the project, using PyScript, or anything else, please ask in the [PyScript forum](https://community.anaconda.cloud/c/tech-topics/pyscript).
 
-## Places to start
+## Code Contributions
+
+### Opening a Pull Request
+
+To contribute code changes or bug fixes to PyScript, follow these steps to open a pull request:
+
+1. Fork the PyScript repository to your GitHub account.
+2. Create a new branch from the `main` branch with a descriptive name for your changes.
```

Comment:
nit: from the `X` branch you are using or testing

Reply:
I've changed to 

> Create a new branch from the `main` branch with a descriptive name for your changes. (Replace `main` with another name if you are starting or testing from another branch)

but I was very debated on this one. Almost feels like that is an exception use case and if you are doing that you have enough experience about "the right thing to do". Keep it as it was, seemed simpler to follow for a newcomer...

Reply:
agreed but most development these days is done in *next* so until we move away core from the rest or we don't publish the whole project as main the *main* branch should be used only for "legacy issues".

Reply:
Yeah. I've changed the wording. Think it should capture the point you are making. Lmk  :)